request-example.go

```diff
@@ -66,9 +66,21 @@ func (fs *root) Filewrite(r *Request) (io.WriterAt, error) {
 		file = newMemFile(r.Filepath, false)
 		fs.files[r.Filepath] = file
 	}
+	return file, nil
+}
+
+func (fs *root) Filewrite(r *Request) (io.WriterAt, error) {
+	file, err := fs.getFileForWrite(r)
+	if err != nil {
+		return nil, err
+	}
 	return file.WriterAt()
 }
 
+func (fs *root) OpenFile(r *Request) (WriterAtReaderAt, error) {
+	return fs.getFileForWrite(r)
```

Comment:
Implement this function with:
```
file, err := fs.getFileForWrite(r)
if err != nil {
  return nil, err
}
return file, nil
```

And then you don’t have to do the hacky check for `&& c != nil && reflect.ValueOf(c).IsNil()`.